src/Admin/SiteHealth.php

```diff
@@ -49,7 +49,7 @@ final class SiteHealth implements Service, Registerable, Delayed, Conditional {
 	 * @return bool Whether the conditional object is needed.
 	 */
 	public static function is_needed() {
-		return is_admin() && ! wp_doing_ajax();
+		return ( ! wp_doing_ajax() );
```

Comment:
Why the need for removing the `is_admin()` check?

Reply:
`is_admin()` check need to remove in order to hook a function on `template_redirect` action. Which is used to render random strings on the front-end.

Reply:
I see. Then I would recommend ensuring that the Site Health checks and related filters are only registered when `is_admin()` is true, and also to only register the `template_redirect` action when the user has the capability to view the Site Health screen.

Reply:
Suggestion applied.